rfc/1216-resolve-cloud-tokens.md

```diff
@@ -0,0 +1,845 @@
+```
+
+When you run `cdk8s synth`, your custom logic will be invoked, and the return 
+value will replace the original value and be written to the manifest. For example, if 
+you define a Kubernetes service like so:
+
+```ts
+new KubeService(this, 'Service', {
+  spec: {
+    type: 'LoadBalancer',
+  }
+});
+```
+
+Your resolver will be invoked with the following arguments:
+
+- **context**
+  - *obj*: The `KubeService` instance of type `ApiObject`.
+  - *key*: `['spec', 'type']`
+- **value**: `LoadBalancer`
+
+
+One common use-case for this feature is to automatically resolve deploy time 
+attributes of cloud resources, and pass them to your Kubernetes workloads. To that end, 
+cdk8s provides a resolver for the following CDK frameworks:
+
+#### AWS Cloud Development Kit
+
+In this example, we create an S3 `Bucket` with the AWS CDK, and pass its (deploy time generated) name 
+as an environment variable to a Kubernetes `CronJob` resource.
+
+```ts
+import * as aws from 'aws-cdk-lib';
+import * as k8s from 'cdk8s';
+import * as kplus from 'cdk8s-plus-26';
+
+import { AwsCdkTokenResolver } from '@cdk8s/aws-cdk-token-resolver';
+
+const awsApp = new aws.App();
+const stack = new aws.Stack(awsApp, 'Stack');
+
+const resolver = new AwsCdkTokenResolver(stack);
+
+const k8sApp = new k8s.App();
+const manifest = new k8s.Chart(k8sApp, 'Manifest', { resolver });
+
+const bucket = new aws.aws_s3.Bucket(stack, 'Bucket');
+
+new kplus.CronJob(manifest, 'CronJob', {
+  schedule: k8s.Cron.daily(),
+  containers: [{
+    image: 'job',
+    envVariables: {
+      // passing the bucket name via an env variable
+      BUCKET_NAME: kplus.EnvValue.fromValue(bucket.bucketName),
+    }
+ }]
+});
+
+awsApp.synth();
+k8sApp.synth();
+```
+
+Notice we create two applications: one for our cdk8s constructs, and one for our AWS CDK constructs.
+Both are defined and synthesized in the same file, but can be separated as needed.
+Since your Kubernetes resources now depend on AWS CDK deploy tokens, you'll first need to run `cdk deploy`, 
+and only then `cdk8s synth`.
+
+> Otherwise, the Kubernetes manifests will contain a string representation of the tokens (e.g `${Token[TOKEN.25]}`), 
+> instead of the concrete values. To learn more AWS CDK tokens, see [here](https://docs.aws.amazon.com/cdk/v2/guide/tokens.html).
+
+The implementation of the resolver utilizes [Cloud Control](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/what-is-cloudcontrolapi.html)
+to fetch the runtime attributes of a resource. However, not all CloudFormation resources are currently supported by Cloud Control.
+
+> For a list of supported resources, see [here](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/supported-resources.html).
+
+In case you need an attribute for such a resource, you can add a `CfnOutput` resource to your stack, and use the resolver to 
+explicitly fetch its value from the deployed stack. For example:
```

Comment:
This feels like typically the kind of thing the resolver should take care of transparently, no? 

If this is more reliable, why don't we pass everything via `CfnOutput` automatically?